packages/flutter/lib/src/widgets/animated_child_switcher.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/animation.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'ticker_provider.dart';
+import 'transitions.dart';
+
+class _AnimatedChildSwitcherChildEntry {
+  _AnimatedChildSwitcherChildEntry(this.widget, this.controller, this.animation);
+
+  Widget widget;
+
+  final AnimationController controller;
+  final Animation<double> animation;
+}
+
+/// A widget that automatically does a [FadeTransition] between a new widget and
+/// the widget previously set on the [AnimatedChildSwitcher] as a child.
+///
+/// More than one previous child can exist and be fading out while the newest
+/// one is fading in if they are swapped fast enough (i.e. before [duration]
+/// elapses).
+///
+/// See also:
+///
+///  * [AnimatedCrossFade], which only fades between two children, but also
+///    interpolates their sizes, and is reversible.
+///  * [FadeTransition] which [AnimatedChildSwitcher] uses to perform the transition.
+class AnimatedChildSwitcher extends StatefulWidget {
+  const AnimatedChildSwitcher({
```

Comment:
Constructor doc: which parameters must be non-null

Reply:
Done.